### ISSUE TITLE
Can do filter with startswith filter?

### ISSUE THREAD

**Issue by mithun2003** [igorbenav/fastcrud#83]
events = await crud_events.get_multi(
        db,
        offset=compute_offset(page, items_per_page),
        return_as_model=True,
        limit=items_per_page,
        sort_columns="id",
        sort_orders="desc",
        schema_to_select=EventRead,
        **conditions,
    ) I want to filter data that are starting user_id

**Comment by mithun2003** [comment]
I think we can achieve that by adding 
`
    def __init__(
        self,
        model: type[ModelType],
        is_deleted_column: str = "is_deleted",
        deleted_at_column: str = "deleted_at",
        updated_at_column: str = "updated_at",
    ) -> None:
        self.model = model
        self.is_deleted_column = is_deleted_column
        self.deleted_at_column = deleted_at_column
        self.updated_at_column = updated_at_column

    def _parse_filters(
        self, model: Optional[Union[type[ModelType], AliasedClass]] = None, **kwargs
    ) -> list[BinaryExpression]:
        model = model or self.model
        filters = []
        for key, value in kwargs.items():
            if "__" in key:
                field_name, op = key.rsplit("__", 1)
                column = getattr(model, field_name, None)
                if column is None:
                    raise ValueError(f"Invalid filter column: {field_name}")

                if op == "gt":
                    filters.append(column > value)
                elif op == "lt":
                    filters.append(column < value)
                elif op == "gte":
                    filters.append(column >= value)
                elif op == "lte":
                    filters.append(column <= value)
                elif op == "ne":
                    filters.append(column != value)
                elif op == "in":
                    if not isinstance(value, (tuple, list, set)):
                        raise ValueError("in filter must be tuple, list or set")
                    filters.append(column.in_(value))
                elif op == "not_in":
                    if not isinstance(value, (tuple, list, set)):
                        raise ValueError("in filter must be tuple, list or set")
                    filters.append(column.not_in(value))
               **elif op == "startswith:
                      filters.appen(column.startswith(value))**
            else:
                column = getattr(model, key, None)
                if column is not None:
                    filters.append(column == value)

        return filters
`

**Comment by JakNowy** [comment]
This is going to be coverred by https://github.com/igorbenav/fastcrud/issues/79

**Comment by mithun2003** [comment]
for get the total count in get_multi_joined instead of using this 
`total_count: int = await self.count(db=db, joins_config=join_config, **kwargs) ` use
` total_count: int = await self.count(db=db, joins_config=join_definitions, **kwargs)`

**Comment by mithun2003** [comment]
"I have two tables, 'user' and 'request'. I want to retrieve the 'name', 'email', and 'score' from the 'user' table, and 'id', 'user_id', and 'created_at' from the 'request' table where 'user.user_id' equals 'request.user_id', 'user.platform_id' equals 'request.platform_id', and 'user.platform_id' equals 2. The main requirement is that if there are multiple users with the same 'user_id', only the data from the user who was created last should be shown. How can I structure the query to achieve this?"

**Comment by igorbenav** [comment]
I didn't test and don't know if this will work, but maybe something like

```python
from sqlalchemy import func, and_
from sqlalchemy.orm import aliased
from sqlalchemy.sql.expression import select, text
from fastcrud import FastCRUD, JoinConfig

# Define a subquery for the latest users by user_id
latest_user_subquery = (
    select([
        User.user_id,
        func.row_number().over(
            partition_by=User.user_id, 
            order_by=User.created_at.desc()
        ).label('rn')
    ]).where(User.platform_id == 2).alias('latest_users')
)

user_crud = FastCRUD(User)

# Join the latest user details with the request table, filtering to get only the latest users
join_config = JoinConfig(
    model=Request,
    join_on=and_(
        User.user_id == Request.user_id, 
        latest_user_subquery.c.user_id == User.user_id, 
        latest_user_subquery.c.rn == 1
    ),
    join_type="inner",
    schema_to_select=RequestReadSchema
)

# Execute the joined query
result = await user_crud.get_multi_joined(
    db=db_session,
    schema_to_select=UserReadSchema,
    joins_config=[join_config],
    platform_id=2
)
```

But at this point you might just use raw sqlalchemy:

```python
from sqlalchemy import select, func
from sqlalchemy.sql import table, column
from sqlalchemy.orm import aliased


# Subquery to get the latest user entries by user_id for platform_id 2
user_subq = (
    select(
        User.user_id,
        User.name,
        User.email,
        User.score,
        func.row_number().over(
            partition_by=User.user_id,
            order_by=User.created_at.desc()
        ).label('rn')
    )
    .where(User.platform_id == 2)
    .alias('latest_users')
)

# Main query that joins the subquery with the request table
result_query = (
    select(
        user_subq.c.name,
        user_subq.c.email,
        user_subq.c.score,
        Request.id.label('request_id'),
        Request.user_id.label('request_user_id'),
        Request.created_at.label('request_created_at')
    )
    .join(
        Request,
        (user_subq.c.user_id == Request.user_id) &
        (user_subq.c.rn == 1) &
        (Request.platform_id == 2)
    )
)

result = session.execute(result_query).fetchall()
```

Again, not tested.

**Comment by igorbenav** [comment]
Closed by #85
